lms/services/assignment.py

```diff
@@ -50,6 +50,12 @@ def create_assignment(self, tool_consumer_instance_guid, resource_link_id):
 
     def update_assignment(self, request, assignment, document_url, group_set_id):
         """Update an existing assignment."""
+        if request.GET.get("learner_canvas_user_id"):
```

Comment:
The fact that having a `learner_canvas_user_id` param implies we are in the context of speed grader is not immediately obvious without having the knowledge or reading the comment below, so I would suggest creating a helper method or function which receives the request and based on it determines if it's a speed grader context.

```python
def is_canvas_speed_grader(request: Request) -> bool:
    return request.GET.get("learner_canvas_user_id")
```

Reply:
Yes, you are right, I made a note about this because we use it in a few places and it could use that method in a common location.

I'll merge this and deal with that in  separate PR.